utils/sysinfo/sysinfo.go

```diff
@@ -464,12 +465,49 @@ func getCoresInfo(sysFs sysfs.SysFs, cpuDirs []string) ([]info.Core, error) {
 			return nil, err
 		}
 
+		var rawBookID, rawDrawerID string
+		var bookID, drawerID int
+		// s390/s390x additional cpu topology levels
+		if runtime.GOARCH == "s390x" {
+			rawBookID, err = sysFs.GetBookID(cpuDir)
+			if os.IsNotExist(err) {
+				klog.Warningf("Cannot read book id for %s, book_id file does not exist, err: %s", cpuDir, err)
+				continue
+			} else if err != nil {
+				return nil, err
+			}
+			bookID, err = strconv.Atoi(rawBookID)
+			if err != nil {
+				return nil, err
+			}
+
+			rawDrawerID, err = sysFs.GetDrawerID(cpuDir)
+			if os.IsNotExist(err) {
+				klog.Warningf("Cannot read drawer id for %s, drawer_id file does not exist, err: %s", cpuDir, err)
+				continue
+			} else if err != nil {
+				return nil, err
+			}
+			drawerID, err = strconv.Atoi(rawDrawerID)
+			if err != nil {
+				return nil, err
+			}
+		}
+
 		coreIDx := -1
 		for id, core := range cores {
-			if core.Id == physicalID && core.SocketID == physicalPackageID {
-				coreIDx = id
+			if runtime.GOARCH == "s390x" {
+				if core.Id == physicalID && core.SocketID == physicalPackageID &&
+					core.BookID == bookID && core.DrawerID == drawerID {
+					coreIDx = id
+				}
+			} else {
+				if core.Id == physicalID && core.SocketID == physicalPackageID {
+					coreIDx = id
+				}
```

Comment:
I think this can be simplified:
```
if core.Id == physicalID && core.SocketID == physicalPackageID {
	// For s390x, we need to check the BookID and DrawerID match as well.
	if runtime.GOARCH != "s390x" || (core.BookID == bookID && core.DrawerID == drawerID) {
		coreIDx = id
	}
}
```

Reply:
Thanks, done.